samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -900,7 +900,7 @@ class SamzaContainer(
 
   def startTask {
     info("Initializing stream tasks.")
-
+    offsetManager.systemAdmins.values.foreach(_.start())
```

Comment:
Since the states and the long-lasting client connections in SystemAdmin are only created when we call SystemAdmin.start(), then it should be OK as of this patch to only call SystemAdmin.start() when we start SamzaContainer, right?

We can have a separate patch to go over all existing usage of SystemAdmin to manage them and call SystemAdmin.start() properly. By splitting the work into two patches, we can reduce the amount of rebase needed for this patch. Does this sound OK?

Reply:
@lindong28 I have concerns on this fast path due to the following reason: ideally, if we want to introduce a concept of life-cycle management on SystemAdmin, we should already clearly define what operations are allowed in each different stage in the life-cycle of the SystemAdmin. For example, no admin operations should be allowed before the SystemAdmin is started. W/ the fast path you propose here, we don't have a clear view of which operation is allowed in which stage any more (i.e. there are some admin operations you can do before start and some admin operations that you can't).